Crash in DomainMatcher.basicMatch

Steps to reproduce:

- Open sample browser
- Navigate to about:blank
- Start typing in the address bar

cc @grigoryk 

```
Process: org.mozilla.samples.browser, PID: 15341
    java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:779)
     Caused by: java.lang.reflect.InvocationTargetException
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:889)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:779) 
     Caused by: java.net.MalformedURLException: unknown protocol: about
        at java.net.URL.<init>(URL.java:596)
        at java.net.URL.<init>(URL.java:486)
        at java.net.URL.<init>(URL.java:435)
        at mozilla.components.support.utils.DomainMatcherKt.basicMatch(DomainMatcher.kt:28)
        at mozilla.components.support.utils.DomainMatcherKt.segmentAwareDomainMatch(DomainMatcher.kt:17)
        at mozilla.components.browser.storage.memory.InMemoryHistoryStorage.getAutocompleteSuggestion(InMemoryHistoryStorage.kt:97)
        at mozilla.components.feature.toolbar.ToolbarAutocompleteFeature.autocomplete(ToolbarAutocompleteFeature.kt:51)
        at mozilla.components.feature.toolbar.ToolbarAutocompleteFeature.access$autocomplete(ToolbarAutocompleteFeature.kt:17)
        at mozilla.components.feature.toolbar.ToolbarAutocompleteFeature$1.invoke(ToolbarAutocompleteFeature.kt:25)
        at mozilla.components.feature.toolbar.ToolbarAutocompleteFeature$1.invoke(ToolbarAutocompleteFeature.kt:17)
        at mozilla.components.browser.toolbar.BrowserToolbar$setAutocompleteListener$1.invoke(BrowserToolbar.kt:167)
        at mozilla.components.browser.toolbar.BrowserToolbar$setAutocompleteListener$1.invoke(BrowserToolbar.kt:49)
        at mozilla.components.ui.autocomplete.InlineAutocompleteEditText$TextChangeListener.afterTextChanged(InlineAutocompleteEditText.kt:555)
        at android.widget.TextView.sendAfterTextChanged(TextView.java:8203)
        at android.widget.TextView$ChangeWatcher.afterTextChanged(TextView.java:10382)
        at android.text.SpannableStringBuilder.sendAfterTextChanged(SpannableStringBuilder.java:1273)
        at android.text.SpannableStringBuilder.replace(SpannableStringBuilder.java:578)
        at android.text.SpannableStringBuilder.replace(SpannableStringBuilder.java:508)
        at android.text.SpannableStringBuilder.replace(SpannableStringBuilder.java:507)
        at android.view.inputmethod.BaseInputConnection.replaceText(BaseInputConnection.java:844)
        at android.view.inputmethod.BaseInputConnection.setComposingText(BaseInputConnection.java:617)
        at android.view.inputmethod.InputConnectionWrapper.setComposingText(InputConnectionWrapper.java:134)
        at mozilla.components.ui.autocomplete.InlineAutocompleteEditText$onCreateInputConnection$1.setComposingText(InlineAutocompleteEditText.kt:508)
        at com.android.internal.view.IInputConnectionWrapper.executeMessage(IInputConnectionWrapper.java:408)
        at com.android.internal.view.IInputConnectionWrapper$MyHandler.handleMessage(IInputConnectionWrapper.java:93)
        at android.os.Handler.dispatchMessage(Handler.java:102)
        at android.os.Looper.loop(Looper.java:154)
        at android.app.ActivityThread.main(ActivityThread.java:6121)
```

--- a/components/support/utils/src/main/java/mozilla/components/support/utils/DomainMatcher.kt
+++ b/components/support/utils/src/main/java/mozilla/components/support/utils/DomainMatcher.kt
@@ -5,6 +5,7 @@
 
 package mozilla.components.support.utils
 
+import java.net.MalformedURLException
 import java.net.URL
 
 const val WWW_PREFIX_OFFSET = 4
@@ -25,14 +26,15 @@ private fun basicMatch(query: String, urls: Iterable<String>): String? {
         if (rawUrl.startsWith(query)) {
             return rawUrl
         }
-        val url = URL(rawUrl)
-        if (url.host.startsWith(query)) {
+
+        val host = (try { URL(rawUrl) } catch (e: MalformedURLException) { null })?.host ?: ""
+        if (host.startsWith(query)) {
             return rawUrl
         }
-        val strippedHost = if (url.host.startsWith("www.")) {
-            url.host.substring(WWW_PREFIX_OFFSET)
+        val strippedHost = if (host.startsWith("www.")) {
+            host.substring(WWW_PREFIX_OFFSET)
         } else {
-            url.host
+            host
         }
         if (strippedHost.startsWith(query)) {
             return rawUrl

--- a/components/support/utils/src/test/java/mozilla/components/support/utils/DomainMatcherTest.kt
+++ b/components/support/utils/src/test/java/mozilla/components/support/utils/DomainMatcherTest.kt
@@ -36,5 +36,7 @@ class DomainMatcherTest {
         )
 
         assertNull(segmentAwareDomainMatch("nomatch", urls))
+
+        assertNull(segmentAwareDomainMatch("about:blank", urls))
     }
 }

